share/functions/export.fish

```diff
@@ -10,7 +10,7 @@ function export --description 'Set global variable. Alias for set -gx, made for
                 set -gx $v $$v
             case 2
                 if contains $v[1] PATH CDPATH MANPATH
-                    set -gx $v[1] (string split ":" $v[2])
+                    set -gx $v[1] (string split ":" (string replace " " ":" $v[2]))
```

Comment:
So you're making a choice here to not support $PATH components with spaces.

Let's stop a moment to think about that.

This is a compatibility shim for ease of copy/pasting and such. It's already imperfect (`export PATH="/bin:${PATH}"` won't work because of the `${}`).

You're choosing between paths-with-spaces and the possibility to append with at least a subset of sh-like syntax. Personally, I haven't ever seen a $PATH component with spaces, and given the issues with that in POSIXly shells it's likely to not be used much.

So I'd say the decision is okay. BUT it needs to be at least stated in a code comment. A mention in the command man/help page would be even better, but since we don't have one for export currently, it doesn't need to be added just for this (though this would be appreciated). So please add a comment explaining why this replace was added.